### DISCUSSION TITLE
Size of the Collection

### DISCUSSION THREAD

**Discussion by alexchudnovsky** [pocketbase/pocketbase#1385] [Q&A] [answered]
Hi All,

Sorry if this question has already been asked, I have checked could not find a reference to this

I am new to pocketbase and am trying to understand how can I get the amount of items in a collection, without having to download the entire collection to get the count of items, is this possible ?

I want to implement pagination and need to get the full size of the collection 

Kind Regards,
Alex

**Comment by uteminhduc** [comment, marked as answer]
Hi,
Pagination in getList [JavaScript SDK](https://github.com/pocketbase/js-sdk)
```js
import PocketBase from 'pocketbase';

const pb = new PocketBase('http://127.0.0.1:8090');

...

// fetch a paginated records list
const resultList = await pb.collection('posts').getList(1, 50, {
    filter: 'created >= "2022-01-01 00:00:00" && someFiled1 != someField2',
});
```

```json
{
  "page": 1,
  "perPage": 50,
  "totalItems": 2,
  "items": [
    {
      "id": "ae40239d2bc4477",
      "collectionId": "a98f514eb05f454",
      "collectionName": "posts",
      "updated": "2022-06-25 11:03:50.052",
      "created": "2022-06-25 11:03:35.163",
      "title": "test1"
    },
    {
      "id": "d08dfc4f4d84419",
      "collectionId": "a98f514eb05f454",
      "collectionName": "posts",
      "updated": "2022-06-25 11:03:45.876",
      "created": "2022-06-25 11:03:45.876",
      "title": "test2"
    }
  ]
}
```
Refs: [https://pocketbase.io/docs/api-records/](url)


> I am new to pocketbase and am trying to understand how can I get the amount of items in a collection, without having to download the entire collection to get the count of items, is this possible ?

You can use trick `perPage=1` to get number of items in collection

**Reply by alexchudnovsky** [reply]
Thank you very much, that makes sense now 

Kind Regards,
Alex
